cohort-engine/src/main/java/com/ibm/cohort/engine/measure/CDMMeasureEvaluation.java

```diff
@@ -108,9 +113,10 @@ public CDMMeasureEvaluation(DataProvider provider, Interval measurementPeriod) {
 	 *                  measure evaluation
 	 * @param patientId Patient ID of the patient to evaluate
 	 * @param evidenceOptions MeasureEvidenceOptions to indicate whether or not to return evaluated resources and define level results
+     * @param parameterMap Map of parameter names to Parameter objects                       	   
 	 * @return MeasureReport with population components filled out.
 	 */
-	public MeasureReport evaluatePatientMeasure(Measure measure, Context context, String patientId, MeasureEvidenceOptions evidenceOptions) {
+	public MeasureReport evaluatePatientMeasure(Measure measure, Context context, String patientId, MeasureEvidenceOptions evidenceOptions, Map<String, Parameter> parameterMap) {
```

Comment:
Passing in the parameters was done because I was unable to figure out how to access all of the parameters in the Context object without knowing the names of each parameter.  I wanted to cover the case where a parameter is passed in through Java, but not actually specified on the Measure.

I'm not sure if that case will happen in production, so I'd also be fine only  outputting parameters that were specified on the input Measure if folks think that would make more sense. In that case,  I would remove this change.

Reply:
Does the parameters map that you added as input contain the parameters that were defined on the Measure resource? I don't think it does. See MeasureEvaluationSeeder.create() for where the parameters defined on the measure report get copied into the Context.

Reply:
It doesn't, but they are on the Measure itself and are retrieved later.